interp/impargs.ml

```diff
@@ -527,7 +531,7 @@ let implicits_of_global ref =
   with Not_found -> [DefaultImpArgs,[]]
 
 let cache_implicits_decl (ref,imps) =
-  implicits_table := GlobRef.Map.add ref imps !implicits_table
+  States.modify_state implicits_tag (fun s -> GlobRef.Map.add ref imps s)
```

Comment:
That certainly looks strange [and likely wrong], as this module is not the owner of the state.

What is the point of doing that indeed?

Reply:
This is correct afaij but a convoluted (and terribly inefficient) way to do the same as the previous line, since, instead of modifying the `implicits_table` reference directly, we freeze the whole state (which includes reading the value `implicits_table`), we modify the `implicits_table` field of this whole state, and we unfreeze, i.e. assign the new value back to the reference `implicits_table`.

There is no point to do that at the current time. It was an experiment to test how much the individual states could be seen as fields of a global state (rather than the global state as a collection of the freeze of the private states).

Note that in #8600, it is not anymore like this.

Reply:
Indeed I see no point on doing that until you return the modified state to the upper layers. Owner is vernacentries so it must reach that part I think.